modules/sdk-coin-avaxp/src/avaxp.ts

```diff
@@ -83,18 +85,26 @@ export class AvaxP extends BaseCoin {
     return true;
   }
 
-  verifyAddress(params: VerifyAddressOptions): boolean {
-    const bintools = BinTools.getInstance();
-    const address = bintools.parseAddress(params.address, 'P');
-    return address !== undefined;
-  }
-
   /**
    * Check if address is valid, then make sure it matches the root address.
    *
    * @param {VerifyAddressOptions} params address and rootAddress to verify
    */
   isWalletAddress(params: VerifyAddressOptions): boolean {
+    const { address, keychains } = params;
+    if (!this.isValidAddress(address)) {
+      throw new InvalidAddressError(`invalid address: ${address}`);
+    }
+    if (!keychains || keychains.length !== 3) {
+      throw new Error('Invalid keychains');
+    }
+    const walletAddress = keychains.map((keychain) =>
+      new AvaxpLib.KeyPair({ pub: keychain.pub }).getAddress(this._staticsCoin.network.type)
+    );
+
+    if (!walletAddress.some((add) => add.endsWith(address))) {
```

Comment:
why is `endsWith` needed here? shouldn't they be equal?

Reply:
The addresses were initially missing the "P-". I think that it changes. I'll check if I could improve it,

Reply:
done